File: frontend/src/types/notification.ts
export interface Notification {
  id: string;
  userId?: string;
  notificationType?: string;
  title: string;
  content?: string;
  relatedEntityType?: string;
  relatedEntityId?: string;
  isRead?: boolean;
  readAt?: string;
  createdAt?: string;
}

export interface NotificationListResponse {
  notifications: Notification[];
  total: number;
  unreadCount: number;
  page: number;
  size: number;
}

export interface NotificationMarkReadResponse {
  notification: Notification;
}

export interface NotificationMarkAllReadResponse {
  count: number;
}

File: backend/src/main/java/com/giapha/model/entity/Notification.java
package com.giapha.model.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Notification {
    private UUID id;
    private UUID userId;
    private String notificationType;
    private String title;
    private String content;
    private String relatedEntityType;
    private UUID relatedEntityId;
    private Boolean isRead;
    private OffsetDateTime readAt;
    private OffsetDateTime createdAt;
}